components/ucd/category/src/lib.rs

```diff
@@ -0,0 +1,313 @@
+    ///
+    /// Abbreviated: Zs
+    SpaceSeparator,
+
+    /// U+2028 LINE SEPARATOR only
+    ///
+    /// Abbreviated: Zl
+    LineSeparator,
+
+    /// U+2029 PARAGRAPH SEPARATOR only
+    ///
+    /// Abbreviated: Zp
+    ParagraphSeparator,
+
+    /// A C0 or C1 control code
+    ///
+    /// Abbreviated: Cc
+    Control,
+
+    /// A format control character
+    ///
+    /// Abbreviated: Cf
+    Format,
+
+    /// A surrogate code point
+    ///
+    /// Abbreviated: Cs
+    Surrogate,
+
+    /// A private-use character
+    ///
+    /// Abbreviated: Co
+    PrivateUse,
+
+    /// Unassigned
+    ///
+    /// Abbreviated: Cn
+    Unassigned,
+}
+
+
+use self::GeneralCategory::*;
+
+// TODO: potentially change these to match pattern matching.
+// Static array contains was easier and clearer to get the first draft up,
+// but is it optimized to the same thing?
+impl GeneralCategory {
+    /// Lu | Ll | Lt
+    ///
+    /// Abbreviated: LC
+    pub fn is_cased_letter(&self) -> bool {
+        [UppercaseLetter, LowercaseLetter, TitlecaseLetter].contains(self)
+    }
+
+    /// Lu | Ll | Lt | Lm | Lo
+    ///
+    /// Abbreviated: L
+    pub fn is_letter(&self) -> bool {
+        [UppercaseLetter, LowercaseLetter, TitlecaseLetter, ModifierLetter, OtherLetter]
+            .contains(self)
+    }
+
+    /// Mn | Mc | Me
+    ///
+    /// Abbreviated: M
+    pub fn is_mark(&self) -> bool {
+        [NonspacingMark, SpacingMark, EnclosingMark].contains(self)
+    }
+
+    /// Nd | Nl | No
+    ///
+    /// Abbreviated: N
+    pub fn is_number(&self) -> bool {
+        [DecimalNumber, LetterNumber, OtherNumber].contains(self)
+    }
+
+    /// Pc | Pd | Ps | Pe | Pi | Pf | Po
+    ///
+    /// Abbreviated: P
+    pub fn is_punctuation(&self) -> bool {
+        [
+            ConnectorPunctuation,
+            DashPunctuation,
+            OpenPunctuation,
+            ClosePunctuation,
+            InitialPunctuation,
+            FinalPunctuation,
+            OtherPunctuation,
+        ].contains(self)
+    }
+
+    /// Sm | Sc | Sk | So
+    ///
+    /// Abbreviated: S
+    pub fn is_symbol(&self) -> bool {
+        [MathSymbol, CurrencySymbol, ModifierLetter, OtherSymbol].contains(self)
+    }
+
+    /// Zs | Zl | Zp
+    ///
+    /// Abbreviated: Z
+    pub fn is_separator(&self) -> bool {
+        [SpaceSeparator, LineSeparator, ParagraphSeparator].contains(self)
+    }
+
+    /// Cc | Cf | Cs | Co | Cn
+    ///
+    /// Abbreviated: C
+    pub fn is_other(&self) -> bool {
+        [Control, Format, Surrogate, PrivateUse, Unassigned].contains(self)
+    }
+}
+
+const GENERAL_CATEGORY_TABLE: &'static [(char, char, GeneralCategory)] =
+    include!("tables/general_category.rsv");
+
+impl GeneralCategory {
+    /// Find the GeneralCategory of a single char.
+    pub fn of(ch: char) -> GeneralCategory {
+        bsearch_range_value_table(ch, GENERAL_CATEGORY_TABLE)
+    }
+}
+
+fn bsearch_range_value_table(
```

Comment:
I'm actually experimenting with porting the python scripts into Rust (to use in cargo buildscripts), partially because I'm just interested, and partially because it could potentially work. And I'm enjoying it (lets me exercise understanding Python code and get more practice writing Rust, so it's not wasted effort. And it's a problem space I wanted to dip my toes into.

Assume rucd works for the generation, it'd definitely be best to use that, once its ready.